internal/rpc/core/mempool.go

```diff
@@ -151,3 +151,7 @@ func (env *Environment) CheckTx(ctx *rpctypes.Context, tx types.Tx) (*coretypes.
 	}
 	return &coretypes.ResultCheckTx{ResponseCheckTx: *res}, nil
 }
+
+func (env *Environment) RemoveTx(ctx *rpctypes.Context, txkey types.TxKey) error {
+	return env.Mempool.RemoveTxByKey(txkey)
```

Comment:
Any reason not to pass through the context into this function? Seems possible clients will want to timeout while calling this code and may even timeout before we get the mempool lock.

Reply:
the mempool interface is internal so it feels totally safe to be able to defer this at the moment.

Reply:
What I'm not sure on is if we should still remove from the mempool even if the ctx has timed out by the time we get the lock. I think it should be fine, but something worth considering.

Reply:
I'm not super worried about this.